### ISSUE TITLE
[BUG]: It has become hard to build Rust code with older compilers

### ISSUE THREAD

**Issue by RalfJung** [compiler-explorer/compiler-explorer#5349]
### Describe the bug

When I just quickly want to build some Rust code with older compilers, that no longer works:
```
<Compilation failed>

# For more information see the output window
# To open the output window, click or drag the "Output" icon at the bottom of this window
```
The output window says
```
error: argument for --edition must be one of: 2015|2018. (instead was `2021`)

Compiler returned: 1
```

### Steps to reproduce

1. visit https://rust.godbolt.org/
2. select an old compiler (I used 1.37.0)

### Expected behavior

This should build successfully, as the given code is compatible with that compiler.

### Reproduction link

Not applicablehttps://rust.godbolt.org/

### Screenshots

Not applicable

**Comment by RalfJung** [comment]
This got introduced via https://github.com/compiler-explorer/compiler-explorer/issues/3765. I found away to work around this, but I literally had to first dig out https://github.com/compiler-explorer/compiler-explorer/issues/3765 to discover that: I have to click the wrench and manually select edition 2015.

I don't think I would ever have found this out by myself. I tried `--edition 2015` in the flags, but that just gave another error (complaining that the edition is now set twice). So while a work-around for this issue exist, I think many people will be stumped and unable to find it.

**Comment by ojeda** [comment]
Yeah, this was my concern and why the proposal I made on https://github.com/compiler-explorer/compiler-explorer/issues/3765 was about pre-filling the flags box rather than doing something implicit. The actual change happened at https://github.com/compiler-explorer/compiler-explorer/pull/5001.

**Comment by dkm** [comment]
I must admit that I had the exact same problem last week.
Maybe we should have a property attached to the compiler to set its default edition, instead of having only one global default.

**Comment by Alcaro** [comment]
If I want to compile something with rustc 1.30, I'll create a new rustc (which will be the default version, which is latest, and will therefore get its default edition), then change it to 1.30. If I'm reading your proposal correctly, that'll do something silly.

The best solution I can think of would be add a Latest value in that edition field, and map that value to different things between editions.

(Another option is auto adjusting the edition if rustc version is changed to something that doesn't support that edition, but that means 'create rustc, change version to 1.70, change to 1.71' and 'create rustc, change version to 1.30, change to 1.71' will unexpectedly yield different results.)

**Comment by dkm** [comment]
By compiler, I meant the actual compiler in the properties files, not the compiler pane. But maybe you're correct, I would need to have a look at how overrides are implemented to be more confident :)
